atlp-rwanda/ecommerce-app-predators-bn
# #16  Ft seller-delete-specific-item

#### What does this PR do?
This pull request adds a new function deleteSpecificProduct to the product API, which allows a seller to delete a specific product by ID. The function takes in the ID of the product to be deleted, as well as a reason for the deletion.

#### Description of Task to be completed?
As a Seller, I should be able to delete a specific item in my colletion
Such that I can remove items I no longer deal in/were errornously added
Given a delete request to delete a specific item in a given seller's collection

#### How should this be manually tested?
1. Verify that the API endpoint for deleting a specific product by ID is working correctly and that the endpoint is accessible.
2. Using a tool like Postman or cURL, make a DELETE request to the API endpoint for deleting a specific product, passing in the ID of a product that exists in the database and a reason for deletion in the request body.
3. Verify that the API returns a 200 success response with a JSend object in the following format:

`{
  "status": "success"
,
  "data": 
{
    "message": "This product has been removed because of the following reason: [reason]."
  }
}`
4. Check that the product has been deleted from the database by making a GET request to the endpoint for retrieving a list of all products and verifying that the deleted product is no longer in the list.
#### Any background context you want to provide?
The function first checks if the product with the given ID exists in the database. If it does not, the function returns a 404 error with a message indicating that the product could not be found. If the product is found, the function deletes it from the database and returns a 200 success response with a message indicating the reason for deletion.

#### What are the relevant pivotal tracker/Trello stories?
cardNumber : #16
Trellolink: https://trello.com/c/WrmHrbaY/16-a-seller-should-be-able-to-delete-a-specific-item

#### Screenshots (if appropriate)
N/A
